src/FieldMapBrBz.cpp

```diff
@@ -20,10 +20,29 @@ namespace fs = std::filesystem;
 
 using namespace dd4hep;
 
+// In DD4hep 1.26, the name for the `field_type` enum changed (from `type`).
+#if DD4HEP_VERSION_GE(1, 26)
+#define DD4HEP_FIELD_TYPE field_type
+#else
+#define DD4HEP_FIELD_TYPE type
+#endif
+
+// This allows us to specify the name of the variable by hand, when patching
+// the previous versions, by setting `DD4HEP_FIELD_TYPE_OVERRIDE`.
+// TODO remove after DD4hep 1.26 release
+#define XSTR(x) STR(x)
+#define STR(x) #x
```

Comment:
Does this section, in principle, need some #ifdef XSTR #undef XSTR ... etc., to minimize side effects?

Reply:
And push pop the previous value? https://gcc.gnu.org/onlinedocs/gcc/Push_002fPop-Macro-Pragmas.html

Reply:
Not needed in cpp file